kafka-rest/src/main/java/io/confluent/kafkarest/ratelimit/FixedCostRateLimitRequestFilter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.ratelimit;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+
+/**
+ * A {@link ContainerRequestFilter} that automatically applies a request rate-limit at a fixed cost
+ * based on the resource/method being requested, according to the {@link
+ * io.confluent.kafkarest.KafkaRestConfig#RATE_LIMIT_COSTS_CONFIG} and {@link
+ * io.confluent.kafkarest.KafkaRestConfig#RATE_LIMIT_DEFAULT_COST_CONFIG} configs.
+ */
+final class FixedCostRateLimitRequestFilter implements ContainerRequestFilter {
```

Comment:
I like how you've injected this behaviour using the Filter, nice thinking :)

Reply:
The one thing you liked in the PR is the one thing I removed, sorry :). Removed the injection to avoid the per-request cost of figuring out the cost/rate-limiter. Now this is configured at server start up.